### PR TITLE
Ground Control Procedure endpoint and mock procedure in code

### PR DESCRIPTION
**Pull request by cburnout**
# Description
Creates a `/procedure` and `/procedure/{name}` end point so we can write our procedures in code for now and move them to the ground control FE and BD when we have the time but for now it lets the UI/UX team have an idea of how this data will come in to create the procedurally generated displays.

Fixes # (issue)

#13 

## Bug Fix or Feature

- [ ] Bug
- [ ] Feature
- [ ] Other

# Testing Instructions

- [ ] Do A
- [ ] Then B
- [ ] See if C breaks

## Linting

- [ ] I've linted my code

### CODE REVIEW

**Comment by cburnell** [comment]
Just created this to show how I think the endpoint should look and what the shape of the data should be. I am putting it out for a few reviews because I think this is going to need a few eyes before its actually good. Just realized I need to change the not found response to be actually not found and not what I currently have.

**Review by isaacdenny — APPROVED** [review]
This is prolly as simple as we can get it. Simple for UI team. others feel free to review before merge

File: ARSIS-Telemetry-GroundControl/groundcontrol-api/app/routers/on_server_procedures/mock_procedure.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+class MockProcedure:
+    def __init__(self):
+        self.name = "Mock Procedure"
+        self.summary = "This is a 100% fake procedure that only exists to make sure we can get data through the system"
+
+        suits_image = base64.b64encode(
+            open("app/routers/on_server_procedures/NASASUITS-logo.png", "rb").read()
+        )
+        task_1_list = [
+            {"type": "image", "body": suits_image, "next_step": ""},
```
Comment (Aetheris743) [inline ARSIS-Telemetry-GroundControl/groundcontrol-api/app/routers/on_server_procedures/mock_procedure.py:13]: Are `next_step` and `next_steps` interchangeable?

File: ARSIS-Telemetry-GroundControl/groundcontrol-api/app/routers/procedures.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+in_mem_procedures = {mock_procedure.get_name(): mock_procedure.get_task_list_encoded()}
+
+
+@router.get("/")
+async def procedures():
+    return in_mem_procedures
+
+
+@router.get("/{name}")
+def procedure(name: str):
```
Comment (Aetheris743) [inline ARSIS-Telemetry-GroundControl/groundcontrol-api/app/routers/procedures.py:18]: Should this be an `async` function?

**Review by Aetheris743 — COMMENTED** [review]
Looks good to me. Is there currently code for the unity side of this?

**Comment by cburnell** [comment]
This has the other half https://github.com/BSU-SUITS-Team/ARSIS-6/pull/35

**Pull request merged**